planner/core/plan_cost_ver2.go

```diff
@@ -35,9 +35,15 @@ func GetPlanCost(p PhysicalPlan, taskType property.TaskType, option *PlanCostOpt
 	return getPlanCost(p, taskType, option)
 }
 
+// GenPlanCostTrace define a hook function to customize the cost calculation.
+var GenPlanCostTrace func(p PhysicalPlan, costV costVer2, taskType property.TaskType, option *PlanCostOption) 
```

Comment:
You are passing `costV costVer2` by value instead of by pointer. Looks like that's not expected.

Reply:
This function generate formula and use option.tracer.appendPlanCostDetial to save the formula. And the cost will be calculated at the sametime.
The calculated cost  won't replace the value of costV, just use to check it same with costV.

Reply:
I'm not sure if I understand you correctly, but we usually pass the pointer here, because:
1. Passing the point can avoid copying the entire struct (though that's not a large struct here)
2. Even if you pass the value, you can't prevent modifying `costV`. Because `costVer2.trace` is a pointer and you can still modify the struct pointed by that.

Reply:
Use pointer is OK.
Pass by value means don't want to user to change costV.cost value which is important for optimizer.